src/testing/cluster/storage_checker.zig

```diff
@@ -81,121 +79,95 @@ pub fn StorageCheckerType(comptime Replica: type) type {
-                checksum,
-            });
-
-            // -1 since we never compact op=1.
-            const compactions_index = @divExact(replica.commit_min + 1, half_measure_beat_count) - 1;
-            if (compactions_index == checker.compactions.items.len) {
-                try checker.compactions.append(checksum);
-            } else {
-                const checksum_expect = checker.compactions.items[compactions_index];
-                if (checksum_expect != checksum) {
-                    log.err("{}: replica_compact: mismatch area=grid expect={x:0>32} actual={x:0>32}", .{
-                        replica.replica,
-                        checksum_expect,
-                        checksum,
-                    });
-                    return error.StorageMismatch;
-                }
-            }
-        }
+        pub fn replica_checkpoint(checker: *Self, superblock: *const SuperBlock) !void {
+            const replica_checkpoint_op = superblock.working.vsr_state.commit_min;
 
-        pub fn replica_checkpoint(checker: *Self, replica: *const Replica) !void {
-            const storage = replica.superblock.storage;
-            const working = replica.superblock.working;
-
-            var checkpoint = Checkpoint{
-                .checksum_superblock_manifest = 0,
-                .checksum_superblock_free_set = 0,
-                .checksum_superblock_client_sessions = 0,
-                // TODO State sync: Enable this when proactive client replies sync is implemented.
-                // .checksum_client_replies = checksum_client_replies(storage),
-                .checksum_client_replies = 0,
-                // TODO: State sync: Enable this (again) when proactive table sync is implemented.
-                // .checksum_grid = checksum_grid(replica),
-                .checksum_grid = 0,
-            };
+            var checkpoint_actual = std.enums.EnumMap(CheckpointArea, u128).init(.{
+                .superblock_manifest = checksum_trailer(superblock, .manifest),
+                .superblock_free_set = checksum_trailer(superblock, .free_set),
+                .superblock_client_sessions = checksum_trailer(superblock, .client_sessions),
+            });
 
-            inline for (comptime std.enums.values(vsr.SuperBlockTrailer)) |trailer| {
-                const trailer_size = @field(working, @tagName(trailer) ++ "_size");
-                var copy: u8 = 0;
-                while (copy < constants.superblock_copies) : (copy += 1) {
-                    const trailer_start = trailer.zone().start_for_copy(copy);
-                    @field(checkpoint, "checksum_superblock_" ++ @tagName(trailer)) |=
-                        vsr.checksum(storage.memory[trailer_start..][0..trailer_size]);
-                }
+            const syncing = superblock.working.vsr_state.sync_op_max > 0;
+            if (!syncing) {
+                checkpoint_actual.put(.client_replies, checksum_client_replies(superblock));
```

Comment:
> That's because we don't actually overwrite memory with garbage on a fault, but rather just set a bit in an out-of-band bitset, right?

Correct!

> So we might have to re-think the approach here onece we start doign misdirected writes (which I think would have to modify the memory directly)

I think for misdirected writes we should do something similar, i.e. store the "incorrect but visible" versions in a separate data structure that we consult on reads, while storing the "true but not visible" versions in `storage.memory`. That is, keep `storage.memory` as pristine as possible, to keep checkers simple.